test/unit/transaction-logs.test.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const Logs = require('../../lib/transaction/logs')
+const { test } = require('tap')
+const sinon = require('sinon')
+
+test('Logs tests', (t) => {
+  t.autoend()
+  let logs
+  let agent
+  t.beforeEach(() => {
+    const config = {
+      event_harvest_config: {
+        harvest_limits: {
+          log_event_data: 2
+        }
+      }
+    }
+    agent = {
+      logs: { addBatch: sinon.stub() },
+      config
+    }
+    logs = new Logs(agent)
+  })
+  t.test('should initialize logs storage', (t) => {
+    t.same(logs.storage, [], 'should init storage to empty')
+    t.same(logs.aggregator, agent.logs, 'should create log aggregator')
+    t.equal(logs.maxLimit, 2, 'should set max limit accordingly')
+    t.end()
+  })
+
+  t.test('it should add logs to storage', (t) => {
+    logs.add('line')
+    t.same(logs.storage, ['line'])
+    t.end()
+  })
+
+  t.test('it should not add data to storage if max limit has been met', (t) => {
+    logs.add('line1')
+    logs.add('line2')
+    logs.add('line3')
+    t.same(logs.storage, ['line1', 'line2'])
+    t.end()
+  })
+
+  t.test('it should flush the batch', (t) => {
+    logs.add('line')
+    logs.flush('1.10')
```

Comment:
Oh, priorities are strings? I thought they were only truncated to 6 decimal places upon display. Upon some quick testing, `transaction.priority` seems to be a number. Does this matter?

Reply:
Yea, you're right. This is just a unit test but I can update to be more accurate if you think it's necessary

Reply:
I tend to view unit tests as "authoritative" of what the API should be, so it may help me or perhaps others down the line if the unit test uses the API as intended. So yeah, I would appreciate this change.